lib/services/batch-relayer/batch-relayer.service.ts

```diff
@@ -166,49 +166,67 @@ export class BatchRelayerService {
         return this.fBeetsBarStakingService.encodeLeave(params);
     }
 
-    public encodeJoinPoolAndStakeInMasterChefFarm({
+    public encodeJoinPoolAndStake({
         userAddress,
         pool,
         userData,
         assets,
         maxAmountsIn,
     }: {
         userAddress: string;
-        pool: GqlPoolWeighted | GqlPoolStable;
+        pool: GqlPoolWeighted | GqlPoolStable | GqlPoolMetaStable;
         userData: string;
         assets: string[];
         maxAmountsIn: BigNumberish[];
     }): PoolJoinBatchRelayerContractCallData {
+        const calls: string[] = [];
         const ethIndex = assets.findIndex((asset) => asset === AddressZero);
         const ethAmount = ethIndex !== -1 ? maxAmountsIn[ethIndex] : undefined;
 
-        const vaultEncodedJoinPool = this.vaultEncodeJoinPool({
-            poolId: pool.id,
-            poolKind: 0,
-            sender: userAddress,
-            recipient: this.batchRelayerAddress,
-            joinPoolRequest: {
-                assets,
-                maxAmountsIn,
-                userData,
-                fromInternalBalance: false,
-            },
-            value: ethAmount || Zero,
-            outputReference: this.toChainedReference(0),
-        });
+        calls.push(
+            this.vaultEncodeJoinPool({
+                poolId: pool.id,
+                poolKind: 0,
+                sender: userAddress,
+                recipient: this.batchRelayerAddress,
+                joinPoolRequest: {
+                    assets,
+                    maxAmountsIn,
+                    userData,
+                    fromInternalBalance: false,
+                },
+                value: ethAmount || Zero,
+                outputReference: this.toChainedReference(0),
+            }),
+        );
 
-        const masterChefDeposit = this.masterChefEncodeDeposit({
-            sender: this.batchRelayerAddress,
-            recipient: userAddress,
-            token: pool.address,
-            pid: parseInt(pool.staking!.id),
-            amount: this.toChainedReference(0),
-            outputReference: Zero,
-        });
+        if (pool.staking && pool.staking.type === 'MASTER_CHEF') {
+            calls.push(
+                this.masterChefEncodeDeposit({
+                    sender: this.batchRelayerAddress,
+                    recipient: userAddress,
+                    token: pool.address,
+                    pid: parseInt(pool.staking!.id),
+                    amount: this.toChainedReference(0),
+                    outputReference: Zero,
+                }),
+            );
+        }
+
+        if (pool.staking && pool.staking.type === 'GAUGE') {
```

Comment:
would make this an `else if`, as it can never be both

Reply:
Done!